gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/util/reference/ReferenceVertex.java

```diff
@@ -38,6 +38,10 @@ private ReferenceVertex() {
 
     }
 
+    public ReferenceVertex(final Object id) {
```

Comment:
good point. a "reference" really just needs to have enough information to identify it so that it is attachable back to the graph. the label has never been required to do that and ends up being a bit superfluous for the case where the user wants to construct a reference independently of the graph as in the case:

```java
g.V(1).addE('knows').to(new ReferenceVertex(2))
```

which I believe is allowed in different language variants currently.  Historically speaking, `ReferenceVertex` grew out of a need in OLAP (where the label has more relevance) and then later became the currency of Gremlin Server network serialization. Not sure if this constructor is dangerous for its convenience. I'd be inclined to leave it